difftool/diff/diff_report.go

```diff
@@ -408,28 +408,27 @@ func diffDeposits(coreSnapshot *Result, dn *Result) Status {
 // diffWithdrawals compares *live* withdarawls from the core snapshot with the same from datanode.
 func diffWithdrawals(coreSnapshot *Result, dn *Result) Status {
 	core := coreSnapshot.Withdrawals
-	datanode := []*vega.Withdrawal{}
-	coreWIDs := map[string]struct{}{}
-	for _, w := range core {
-		coreWIDs[w.Id] = struct{}{}
-	}
+	datanode := dn.Withdrawals
 
-	// filter only live withdrawals from datanode
-	for _, w := range dn.Withdrawals {
-		if _, ok := coreWIDs[w.Id]; ok {
-			datanode = append(datanode, w)
-		}
+	// Only compare if withdrawal is both core and datanode; core may have more as it currently
+	// never deletes any, but a datanode without history will initially have none.
+	// Issue about core behavior: https://github.com/vegaprotocol/vega/issues/7440
+	coreByID := map[string]*vega.Withdrawal{}
+	datanodeById := map[string]*vega.Withdrawal{}
+	intersection := map[string]struct{}{}
+
+	for _, w := range coreSnapshot.Withdrawals {
+		coreByID[w.Id] = w
+		intersection[w.Id] = struct{}{}
 	}
 
-	sort.Slice(core, func(i, j int) bool { return core[i].Id < core[j].Id })
-	sort.Slice(datanode, func(i, j int) bool { return datanode[i].Id < datanode[j].Id })
-	if len(core) != len(datanode) {
-		return getSizeMismatchStatus("withdrawals", core, datanode)
+	for _, w := range dn.Withdrawals {
+		datanodeById[w.Id] = w
+		intersection[w.Id] = struct{}{}
 	}
 
-	for i, a := range core {
-		d := datanode[i]
-		if a.String() != d.String() {
+	for id := range intersection {
```

Comment:
isn't the way it's implemented union rather than intersection which is not what you mean?

Reply:
err, that is a very valid point.. 🤦